hooks/playbooks/adoption_deploy_ceph.yml

```diff
@@ -0,0 +1,138 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Deploy ceph for 17.1 deployment
+  hosts: "{{ cifmw_target_host | default('localhost') }}"
+  tasks:
+    - name: Gather ansible_user_dir from undercloud
+      delegate_to: "osp-undercloud-0"
+      ansible.builtin.setup:
+        gather_subset:
+          - user_dir
+
+    - name: Deploy 17.1 ceph
+      vars:
+        stack: >-
+          {{
+            _adoption_source_scenario.stacks[stack_index | int]
+          }}
+        _overcloud_name: >-
```

Comment:
suggestion: I see the overcloud name is coming from [1], you are setting a default here and in other places for `stack.stackname` but it seems all the other vars coming from the scenario are left in this role without default. 

I think it would be best to have the vars defined in one place, the scenario. And I think it would be good to have an assert in the adoption_osp_deploy role that everything mandatory is provided.  

[1] https://github.com/openstack-k8s-operators/data-plane-adoption/blob/43713883a05bb2fdfecb3fb19f24fd594051fc41/scenarios/hci.yaml#L36

Reply:
indeed, this default is a leftover from the beginning that I forgot about, I'll remove it and try to add some validation.